### PR TITLE
[Fix]:  Fixed Navbar dark mode navigation

### LINKED ISSUE #1057: 🐛 Bug: Some Section need's Improvement
### Describe the bug

I encountered an issue with the navbar when dark mode is enabled. I'm unable to see which page I'm currently navigating to because there is no visible active element. Typically, the active route or page is highlighted with a different color, but in dark mode, all the navbar items appear the same.

Screenshot 2024-10-23 at 5 03 19 PM

### Steps To Reproduce

1. Turn on Dark Mode
2. Check the Active Item of Navbar

### Expected Behavior

In dark mode, it's hard to see which page we're currently on. There should be an active color for the navbar items so the current page is clearly visible.

Screenshot 2024-10-23 at 5 03 07 PM

### Are you working on this issue?

Yes

### Do you think this work might require an [Architectural Decision Record (ADR)]? (significant or noteworthy)

NO

### PR DESCRIPTION
**Pull request by officeneerajsaini**
This Pull Request Fixes the Navbar Navigation Color in Dark mode.

**Issue Number:**
-  Closes #1057 

**Screenshots/videos:**
Before:

https://github.com/user-attachments/assets/eeec6f55-fa84-4c9a-83ee-2658141c4594

After : 

https://github.com/user-attachments/assets/9bcd0f82-f0b8-4259-841b-596bf9390f2a

### CODE REVIEW

**Comment by officeneerajsaini** [comment]
In macOS, I'm encountering another issue with the Navbar search button. The search button's style contains a background-color property.

**Comment by officeneerajsaini** [comment]
Screenshot 2024-10-24 at 11 03 04 AM

**Comment by DhairyaMajmudar** [comment]
> I'm sorry for making this pull request complex for the reviewer. Initially, I did my work well, but the `package-lock.json` and `yarn.lock` files were included. Then I made another commit, and I wanted to combine both commits into a single one. Due to some issues, I deleted my branch and then restored it. Now, I've combined both commits into one, but it also includes the previous merged commit. That's what has been happening with me.

No worries with that just remove the changes from `yarn.lock` and `package.json` files.

**Comment by officeneerajsaini** [comment]
> > I'm sorry for making this pull request complex for the reviewer. Initially, I did my work well, but the `package-lock.json` and `yarn.lock` files were included. Then I made another commit, and I wanted to combine both commits into a single one. Due to some issues, I deleted my branch and then restored it. Now, I've combined both commits into one, but it also includes the previous merged commit. That's what has been happening with me.
> 
> No worries with that just remove the changes from `yarn.lock` and `package.json` files.

Yeah I did 😄

**Comment by officeneerajsaini** [comment]
The commit #1049 is also included in my pull request.

File: package.json
Comment (DhairyaMajmudar) [inline package.json:1]: Seems like there are still some more changes left to be removed : )

File: yarn.lock
Comment (DhairyaMajmudar) [inline yarn.lock:1]: From here too

File: package.json
Comment (officeneerajsaini) [reply package.json:1]: What should I do now then, that commit already merged with that code.

**Comment by officeneerajsaini** [comment]
@DhairyaMajmudar Please Review Code now.

**Comment by DhairyaMajmudar** [comment]
@officeneerajsaini pls. don't change the font colors

**Comment by officeneerajsaini** [comment]
> @officeneerajsaini pls. don't change the font colors

In dark mode, some light colors look better than a dark-on-dark combination.

Consistent color:

https://github.com/user-attachments/assets/7fa172a0-1b4d-4dfa-bfe0-1c1f1c9dd76c

Some Lightness :

https://github.com/user-attachments/assets/99eb2397-6a14-4ee3-9945-266d136dfcf8

Should I go with the website theme color?

**Review by benjagm — CHANGES_REQUESTED** [review]
Looks great, Thanks!!

 However I do prefer the white color for the navbar fonts instead that light grey. Can we change that?

**Comment by officeneerajsaini** [comment]
> Looks great, Thanks!!
> 
> However I do prefer the white color for the navbar fonts instead that light grey. Can we change that?

Yes, and what about the color for the active and hover effects?

**Comment by officeneerajsaini** [comment]
https://github.com/user-attachments/assets/9bb68b07-00d7-4f28-8a54-facde004797f

**Comment by officeneerajsaini** [comment]
@benjagm  now take a look on code

**Comment by benjagm** [comment]
It looks better. However, the blue color is difficult to distinguish from the background. Can we find something lighter? Why not just underlying the current section instead of changing the color?

**Comment by officeneerajsaini** [comment]
https://github.com/user-attachments/assets/034906a6-44d7-4623-890c-078fcf5cae46
I chose this color from the hover effect on the 'Star on GitHub' button; it seems to look good.

File: styles/globals.css
Comment (DhairyaMajmudar) [inline styles/globals.css:1]: It seems like this class is no longer used, if it is so you can remove it.

Comment (officeneerajsaini) [reply styles/globals.css:1]: Dhairya, I changed the grey color to white in response to Benjamin's request.

> Looks great, Thanks!!
> 
> However I do prefer the white color for the navbar fonts instead that light grey. Can we change that?

I also added a new CSS property to show an underline on the current page, so some additional CSS properties need to be added. 😄

Comment (officeneerajsaini) [reply styles/globals.css:1]: This class is used when there are 6 community meetings, and events are available.

Without class :
Screenshot 2024-10-23 at 5 00 04 PM

with class :
Screenshot 2024-10-23 at 5 02 02 PM

Comment (DhairyaMajmudar) [reply styles/globals.css:1]: Awsm! With that then keep it don't remove

**Comment by benjagm** [comment]
Hi everyone.

I checked the current version and found this behavior:
- Light style:
  - Style looks good 
  - When navigating using links in the header, all styles to highlight the current section work with the exception of Tools. When in tools there is no style in the header highlight the current version.
- Drak Style:
  - When navigating using links in the header, the styles to highlight the current version don't work.

**Comment by officeneerajsaini** [comment]
> * Light style:
>   
>   * Style looks good
>   * When navigating using links in the header, all styles to highlight the current section work with the exception of Tools. When in tools there is no style in the header highlight the current version.

Yes , Found that part too .

**Comment by benjagm** [comment]
@officeneerajsaini we are very close! Let's make the last change and push it. Great work so far. Thanks for your patience.

**Comment by benjagm** [comment]
Hi @officeneerajsaini . Is there any update on this?

**Comment by benjagm** [comment]
Thanks @officeneerajsaini . Ok I think the reason why the preview deployment I checked is outdated is because the deployment work failed due to some format checks: 

https://github.com/json-schema-org/website/actions/runs/11986321387/job/33419247261?pr=1059

Can you run prettier on that file? I am so looking forward to merge it.

thanks for your patience. It has been a very busy weeks for me on my new job.

**Comment by officeneerajsaini** [comment]
@benjagm Yeah, I know you were very busy at that time, which is why I waited for your response.
I have done everything that was needed. On my local host, everything is working well, but in this preview development, I can't see the underlining navigation.

**Comment by officeneerajsaini** [comment]
Now This Code Ready To Merge @benjagm Please take a Look.

**Review by benjagm — APPROVED** [review]
Thanks for the hard work Neeraj. I just added a small change to make it work. Kudos!

**Comment by officeneerajsaini** [comment]
[Thanks,](url) Benja 🫠

**Pull request merged**